pkg/xds/auth/k8s/authenticator.go

```diff
@@ -38,67 +40,75 @@ func (k *kubeAuthenticator) Authenticate(ctx context.Context, resource model.Res
 }
 
 func (k *kubeAuthenticator) authDataplane(ctx context.Context, dataplane *core_mesh.DataplaneResource, credential auth.Credential) error {
-	if credential == "" {
-		return errors.New("authentication failed: k8s token is missing")
+	proxyName, proxyNamespace, err := util_k8s.CoreNameToK8sName(dataplane.Meta.GetName())
+	if err != nil {
+		return err
+	}
+	serviceAccountName, err := k.podServiceAccountName(ctx, proxyName, proxyNamespace)
+	if err != nil {
+		return errors.Wrap(err, "could not retrieve Pods Service Account to verify identity of a dataplane proxy")
```

Comment:
This error message substantially repeats the one in `podServiceAccountName`. Maybe trim or or the other?

Same in `authZoneIngress`.